poetry/console/commands/shell.py

```diff
@@ -0,0 +1,55 @@
+from os import environ
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        from poetry.config import Config
+
+        # Check if it's already activated or doesn't exist and won't be created
+        if "POETRY_ACTIVE" in environ or not self.venv.is_venv():
+            current_venv = environ.get("VIRTUAL_ENV")
+
+            if current_venv:
```

Comment:
I think we need to do one of two thing here:
1) we assume at that point that `VIRTUAL_ENV` will not be `None` and so this check (and line 23) is unnecessary
2) we handle the case when `current_venv = environ.get("VIRTUAL_ENV")` returns `None`